packages/presentation/lib/src/app/app_bloc.dart

```diff
@@ -107,5 +117,15 @@ class _AppBloc extends BlocImpl<BaseArguments, AppData> implements AppBloc {
   @override
   void bottomBarNavigation(int index) {
     emit(data: tile.copyWith(bottomNavIndex: index));
+    final page = bottomNavBarStack[index]?.call();
+    if (page == null) return;
+    switch (index) {
+      case _homePageKey:
+        _popAllAndPush(page);
+        break;
+      case _loginPageKey:
+        _popAllAndPush(page);
+        break;
```

Comment:
do it via enum. so, u will have the following flow:

u will have a enum for bottom navigation
enum BottomNavigationItemType {
home
...
...
login
}

u can create a static method/extension, which consumes number as parameter and returns enum item (as each enum has its own index

u refactor your code like this

final type = BottomNavigationItemType.fromIndex(index)
switch (type) {
case BottomNavigationItemType.home -> do smth

it seems much prettier